2600hz/kazoo
# [master] Better iso8601

This PR is adding timezone part to ISO 8601. For example CDRs now has a key `iso_8601_combined`, which is formatted version of timestamp in ISO 8601 format.

`kz_time:iso8601/1` used to return date in `2015-04-07T00:00:00` format. This is a correct format based o the standard but lacking the timezone part, it indicate that the date and time is local zone. Since kazoo is always operating in UTC, per spec we have to set timezone. In ISO 8601 timezone for UTC is `+00:00 or `Z` (shorter).

So this function now returns this from now on:

"2015-04-07T00:00:00**Z**"

Supporting for other timezone is added to `kz_time:iso8601/2`. You can pass the name of the timezone or specify time offset `-05:00` or specifying an ingteger indicating seconds (positive for + timezone and negtive for - timezone) like `-18000` for `America/New_York`.

`kz_time:from_iso8601` is added to parse datetime from iso 8601 format. If the string doesn't have timezone it assumes the datetime is in UTC!

Timezone is calculated by converting the time timestamp from UTC to the local zone and calculating its difference from UTC.

### How to test

Observe `iso_8601_combined` in the output of CDRs (in Web Developer for example) to make sure the `Z` is added to the end of the time string.

4.3: https://github.com/2600hz/kazoo/pull/5203
